pkg/sidecar-controller/snapshot_controller.go

```diff
@@ -580,12 +587,19 @@ func (ctrl *csiSnapshotSideCarController) setAnnVolumeSnapshotBeingCreated(conte
 	contentClone := content.DeepCopy()
 	metav1.SetMetaDataAnnotation(&contentClone.ObjectMeta, utils.AnnVolumeSnapshotBeingCreated, "yes")
 
-	updatedContent, err := ctrl.clientset.SnapshotV1().VolumeSnapshotContents().Update(context.TODO(), contentClone, metav1.UpdateOptions{})
+	var patches []utils.PatchOp
+	patches = append(patches, utils.PatchOp{
+		Op:    "replace",
+		Path:  "/metadata/annotations",
+		Value: contentClone.ObjectMeta.GetAnnotations(),
```

Comment:
Since we only need the annotations part, can we call content.ObjectMeta.GetAnnotations() first, save it, and then append the new utils.AnnVolumeSnapshotBeingCreated="yes" annotation to it.  So we don't have to do a DeepCopy on line 587.

Reply:
Or may be do something similar to what you did with finalizers?  https://github.com/kubernetes-csi/external-snapshotter/pull/526/files#diff-1a3473f9016f7abcf5007414ff7141e39032aaf7aba079eeef262d04167208a2R813-R828

Reply:
Good idea, addressed this by creating a new annotations map. I wanted to avoid editing the existing content object, as it's passed up the stack and used when we update the status error.